File: 2021/02/solve.py
#!/usr/bin/python
import os
import sys

path = os.path.join(os.path.dirname(__file__), sys.argv[1] if len(sys.argv) > 1 else 'input')

with open(path, 'r') as f:
    commands = [line.strip().split() for line in f.readlines()]

aim = 0
position = 0
depth = 0

for command, amount in commands:
    if command == 'down':
        aim += int(amount)
    elif command == 'up':
        aim -= int(amount)
    elif command == 'forward':
        position += int(amount)
        depth += aim * int(amount)

answer1 = position * aim
answer2 = position * depth

print('--- Day 2: Dive! ---')
print(' Answer 1: ' + str(answer1))
print(' Answer 2: ' + str(answer2))

File: 2022/01/solve.r
#!/usr/bin/env Rscript

all_args <- commandArgs(trailingOnly=FALSE)
args <- commandArgs(trailingOnly=TRUE)
if (all_args[1] == 'RStudio') {
	library('rstudioapi')
	path <- rstudioapi::getSourceEditorContext()$path
} else {
	path <- tail(c('.', gsub('^(?:|--file=)(.*)$', '\\1', all_args[c(which(all_args == '-f') + 1, which(grepl('^--file=', all_args)))])), 1)
}
path <- file.path(normalizePath(dirname(path)), ifelse(length(args)>0, args[1], 'input'))

calories <- c(0L)

for (line in as.integer(readLines(path))) {
	if (is.na(line)) {
		calories <- c(0L, calories)
	} else {
		calories[1] <- calories[1] + line
	}
}

calories <- sort(calories, TRUE)

answer1 <- calories[1]
answer2 <- sum(calories[1:3])

cat('--- Day 1: Calorie Counting ---\n')
cat(' Answer 1:', answer1, '\n')
cat(' Answer 2:', answer2, '\n')
